CMakeLists.txt

```diff
@@ -0,0 +1,45 @@
+cmake_minimum_required(VERSION 3.20)
+project(libCEED LANGUAGES C CXX)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_EXPORT_COMPILE_COMMANDS 1)
+
+if (CEED_ENABLE_CUDA)
+  enable_language(CUDA)
+endif()
+
+file(GLOB_RECURSE interface_c_sources ${PROJECT_SOURCE_DIR}/interface/*.c)
+file(GLOB_RECURSE interface_cpp_sources ${PROJECT_SOURCE_DIR}/interface/*.cpp)
+
+if (NOT CEED_ENABLE_HIP)
+    list(REMOVE_ITEM interface_c_sources ${PROJECT_SOURCE_DIR}/interface/ceed-hip.c)
+endif()
+
+if (NOT CEED_ENABLE_CUDA)
+    list(REMOVE_ITEM interface_c_sources ${PROJECT_SOURCE_DIR}/interface/ceed-cuda.c)
+endif()
+
+file(GLOB_RECURSE gallery_sources ${PROJECT_SOURCE_DIR}/gallery/*.c)
+
+add_library(CEED ${interface_c_sources} ${interface_cpp_sources} ${gallery_sources})
+add_subdirectory(backends)
+target_include_directories(CEED PUBLIC include)
+target_compile_definitions(CEED PUBLIC CEED_JIT_SOURCE_ROOT_DEFAULT="${PROJECT_SOURCE_DIR}/include")
+
+# TODO: which flags should be enabled for a given platform?
+target_compile_options(CEED PRIVATE "-march=native" 
+                                    "-fvectorize" 
+                                    "-fslp-vectorize" 
+                                    "-ffp-contract=fast")
```

Comment:
I think we shouldn't try to innovate here. We could use the same flags as are in the makefile (but hopefully not by copy/pasting them) or (likely preferable) just delete this.

Reply:
I don't understand, how do you want to mimic the Makefile flags without copying them into the CMake build? Does the Makefile already determine the system hardware and select the appropriate flags automatically?

Reply:
I think he's specifically referring to the `-fvectorize` and `-fslp-vectorize`, which don't appear in our Makefile.

Reply:
Also, the options like `-march=native` are only applied for gcc/clang, not with vendor compilers for which it isn't allowed. I'd rather have nothing here than default options that break. I also don't love the prospect of manually keeping these compatible, but could be convinced it's worthwhile. One could parse `Makefile` or otherwise move those defaults to a place that both CMake and the standard Makefile can pull from.